### DISCUSSION TITLE
Deactivate Isso for certain articles

### DISCUSSION THREAD

**Discussion by Fryboyter** [posativ/isso#908] [Ideas]
Hello everyone.

I am using Isso for several years on various websites. From time to time, however, I publish articles for which I don't want any comments. Be it because I assume that the discussion will drift in an unpleasant direction or because I simply don't want a discussion at all. Or because I am too lazy to moderate the comments.

Because my articles are currently created using Markdown files, I have created a query that checks whether **nocomments: true** is present in the Front Matter section of the article. If so, the code responsible for Isso is not displayed in the article in question and thus the comment function is deactivated. However, because I basically don't know how to code, the code responsible for this probably sucks and may offer loopholes so that you can still create comments.

Therefore, I would like Isso to offer a way to disable the comment function for certain articles. What do you think of this idea?

Because I have almost no programming skills, I unfortunately can't contribute any code to this proposal myself.

Greetings
Fryboyter

**Comment by BBaoVanC** [comment]
I think the best way to do this is in your website code rather than in Isso. I'm not sure what your code looks like, but it should just be an if statement with all the Isso embed code inside.

If it helps at all, you can look at the code on my site (which uses Hugo), where I read the `comments` parameter from the front matter and use it to enable comments (you could just do an "if not" and then use `nocomments` instead in your site to enable comments by default):

- the if statement: https://github.com/BBaoVanC/bobatheme/blob/637ffc12f5da3fc04144dc92ff8399a1a8c59a9e/layouts/_default/single.html#L71-L75
- the template that's included inside that if statement: https://github.com/BBaoVanC/bbaovanc.com/blob/ddb0045088fd9a4aef651da60c4999e3d38d55c8/layouts/partials/comments.html

**Comment by ix5** [comment]
I agree with @BBaoVanC. The correct solution is to disable loading the comment embed script. Feel free to submit a PR to document that.

I _think_ this question has been asked before, though I can't seem to find the issue in the bug tracker.

**Comment by Fryboyter** [comment]
Hello,

Sorry for the late reply, but I'm having a lot of work to do at the moment.

I have already found a solution for my sites like fryboyter.de (for example https://codeberg.org/Fryboyter/hugo/src/branch/main/themes/GythaOgg/layouts/partials/head.html#L35-L46 and https://codeberg.org/Fryboyter/hugo/src/branch/main/themes/GythaOgg/layouts/partials/comments.html).

But this is only a solution for me. Or if the solution is reasonably well programmed, which I wouldn't necessarily assume in my case, for sites that use Hugo. But not for all other sites that use, for example, another static website generator without a comment function. Therefore, in my opinion, a PR would also be of little use.

Because I suspect that Isso is also used by some users who are as bad at programming as I am, or perhaps don't want to or can't program at all. If so, only a precise manual for various generators such as Hugo, Pelikan etc. would be useful. In my opinion, however, this would no longer have much to do with the Isso project. Therefore, a corresponding function of Isso would be the better solution.

All in all, however, I have no problem with the proposal being rejected. In principle, deactivating comments on my pages works. And all in all, I'm glad that projects like Isso exist.

@ix5 
I agree that this has been discussed before. But on one hand, a long time ago. And for another, I have not found a discussion / issue myself.

**Reply by ix5** [reply]
As it currently stands, this wish will not be implemented - it'd make Isso too complicated. The correct way, as stated once before, is to disable loading the embed script.

If we, at some point in the future, decide to make the admin interface more powerful, this is certainly a feature for the wishlist. That feature would also coincide with an existence check for pages - which some people really don't want to have, since they may hide their pages behind a username/password or oauth.

@Fryboyter please use this opportunity to document how you disabled the comments for specific articles and submit a PR for that docs update. Maybe the "Advanced Integration" article is the correct place for that. Isso needs better documentation for integrating with static site generators in general - we should at least cover Pelican, Hugo, Nikola and sphinx.

**Reply by BBaoVanC** [reply]
@ix5 if you could make a page/section for disabling comments on specific pages in the docs, I could add some instructions for how to do it on Hugo (since that's what I use and am familiar with).

**Reply by ix5** [reply]
I'm a bit swamped right now, but feel free to create that page yourself.

**Comment by ix5** [comment]
Btw, this is what I was thinking of: https://github.com/posativ/isso/issues/630

Also related: https://github.com/posativ/isso/issues/679, https://github.com/posativ/isso/issues/71, https://github.com/posativ/isso/issues/180
